packages/relay-runtime/util/recycleNodesInto.js

```diff
@@ -30,29 +30,31 @@ function recycleNodesInto<T>(prevData: T, nextData: T): T {
   const prevArray = Array.isArray(prevData) ? prevData : null;
   const nextArray = Array.isArray(nextData) ? nextData : null;
   if (prevArray && nextArray) {
+    const isFrozen = Object.isFrozen(nextArray);
     canRecycle =
```

Comment:
I might be missing something, but would it make sense to just assign `canRecycle` to `false` if it is frozen?

Reply:
Hey @jstejada, that's a good question. I would *not* want to prematurely decide that `canRecycle` is `false`, since it's still possible for `wasEqual && nextValue === prevArray[ii]` to be true, and we wouldn't want to lose out on the ability to recycle deep objects just because it is frozen.

Reply:
Ah I see, thanks for the explanation!